src/test/ui/issues/issue-41742.rs

```diff
@@ -21,5 +21,7 @@ impl IndexMut<u32> for H {
 }
 
 fn main() {
-    H["?"].f(); //~ ERROR mismatched types
+    H["?"].f();
+    //~^ ERROR mismatched types
+    //~| ERROR mismatched types
```

Comment:
It'd be nice to not cause this duplication of errors, but that's less important than the regression.

Instead of `demand_coerce` you could use `demand_coerce_diag` to be able to `map_err` it and `.delay_span_bug()` it. I _think_ that would be enough, but we should probably not include that in a backport.

Reply:
I am currently seeking a way not to do any demand_coerce at all. If that's not possible (or would be too complex), I'll update the PR to use demand_coerce_diag.